libs/execution_context/ebpf_link.c

```diff
@@ -507,6 +507,30 @@ _ebpf_link_instance_invoke_batch_begin(
     }
     provider_reference_held = true;
 
+    uint8_t maximum_irql;
+    uint8_t minimum_irql;
+
+    return_value = ebpf_program_get_irql_range(link->program, &minimum_irql, &maximum_irql);
+    if (return_value != EBPF_SUCCESS) {
+        EBPF_LOG_MESSAGE_ERROR(
+            EBPF_TRACELOG_LEVEL_ERROR,
+            EBPF_TRACELOG_KEYWORD_LINK,
+            "Query of program maximum IRQL failed.",
+            return_value);
+        goto Done;
+    }
+
+    if ((execution_context_state->current_irql > maximum_irql) ||
```

Comment:
These checks are necessary for the bpf_prog_test_run case, but for the common case, can the program provider be trusted to call eBPF within the correct IRQL range? These checks don't look too expensive, but I wonder if they can be moved into the bpf_prog_test_run code path rather than common?

Reply:
That's a good question. If eBPF core trusts the provider, then specifying the range is redundant. All that is needed is a hint at what level to run the program during bpf_prog_test run.